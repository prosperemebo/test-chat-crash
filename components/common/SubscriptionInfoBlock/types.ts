export interface SubscriptionInfoBlockProps {
    from: 'settings'
    hasMembership: boolean
}