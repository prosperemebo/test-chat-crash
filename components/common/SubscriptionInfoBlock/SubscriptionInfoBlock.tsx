import { useRouter, Href } from 'expo-router'
import analytics from '@/services/analytics'
import {
  SubscriptionDetailContainer,
  LabelText,
  LabelSecondText,
  UnlockBenefitsContainer,
  UnlockBenefitsButton,
  UnlockBenefitsButtonText,
  ButtonBackground,
  StateLayer,
} from './styles'
import i18n from '@/localization/i18n'
import { SubscriptionInfoBlockProps } from './types'


const SubscriptionInfoBlock = (props: SubscriptionInfoBlockProps) => {
  const { from, hasMembership } = props
  const router = useRouter()
  const onPress = () => {
    const to = hasMembership ? '/(app)/settings/subscriptionManagement' : '/(app)/settings/subscription'

    analytics.subscriptions.membershipPageNavigate(from)
    router.push(to as Href)
  }

  const renderContent = () => {
    if (hasMembership) return <LabelText>{i18n.t('subscription:manageSubscriptionsText')}</LabelText>

    return <UnlockBenefitsContainer>
      <LabelText>
        {i18n.t('subscription:unlockAllBenefitsHeader')}
      </LabelText>
      <LabelSecondText>
        {i18n.t('subscription:unlockAllBenefitsDescription')}
      </LabelSecondText>
      <UnlockBenefitsButton onPress={onPress}>
        <UnlockBenefitsButtonText>{i18n.t('from $4.99/mo')}</UnlockBenefitsButtonText>
      </UnlockBenefitsButton>
    </UnlockBenefitsContainer>
  }

  //TODO check for angle of gradient
  return <SubscriptionDetailContainer disabled={!hasMembership} onPress={onPress}>
    <ButtonBackground>
      <StateLayer>
        {renderContent()}
      </StateLayer>
    </ButtonBackground>
  </SubscriptionDetailContainer>
}

export default SubscriptionInfoBlock