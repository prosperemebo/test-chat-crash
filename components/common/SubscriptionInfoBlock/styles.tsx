import styled from 'styled-components/native'
import { LinearGradient } from 'expo-linear-gradient'

type ButtonBackgroundType = {
  colors?: [string, string]
}

export const SubscriptionDetailContainer = styled.TouchableOpacity`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-horizontal: 15px;
  flex-direction: column;
`

export const LabelText = styled.Text`
  text-align: center;
  color: ${props => props.theme.primaryTextColor};
  line-height: 27px;
  font-size: 18px;
  font-family: 'Poppins-Bold';
`

export const LabelSecondText = styled.Text`
  font-family: 'Poppins-Light';
  text-align: center;
  margin-left: 4px;
  color: ${props => props.theme.primaryTextColor};
  line-height: 17px;
  font-size: 12px;
  opacity: 0.8;
  letter-spacing: 0;
`

export const UnlockBenefitsContainer = styled.View`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 250px;
`

export const UnlockBenefitsButton = styled.TouchableOpacity`
  border-radius: 50px;
  border: 1px;
  border-color: ${props => props.theme.secondaryBgLight};
  background-color: ${props => props.theme.secondaryBgLight};
  gap: 1px;
  width: 130px;
  height: 40px;
  margin: 10px;
  padding-top: 10px;
  padding-right: 20px;
  padding-bottom: 10px;
  padding-left: 20px;
`

export const UnlockBenefitsButtonText = styled.Text`
  color: ${props => props.theme.primaryTextColor};
  font-family: 'Poppins-Bold';
  font-size: 12px;
  line-height: 18px;
  text-align: center;
`

export const ButtonBackground = styled(LinearGradient).attrs((props) => ({
  colors: [props.theme.common.gradientButtonColorStart, props.theme.common.gradientButtonColorEnd],
  start: { x: 0, y: 0 },
  end: { x: 1, y: 0 },
}))<ButtonBackgroundType>`
  border-radius: 8px;
  background-color: transparent;
  width: 350px;
  overflow: hidden;
  margin-bottom: 10px;
  justify-content: center;
  align-items: center;
  flex-direction: row;
`

export const StateLayer = styled.View`
  flex-direction: row;
  padding-top: 10px;
  padding-bottom: 10px;
`