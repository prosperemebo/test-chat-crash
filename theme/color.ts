export const WHITE = '#FFFFFF'
export const t10WHITE = 'rgba(255,255,255,0.1)'
export const t76WHITE = 'rgba(255,255,255,0.76)'

export const RED = '#f60004'
export const RED1 = '#D33E41'


export const BLACK = '#000000'
export const BLACK2 = '#050608'
export const nBLACK = '#15161A'
export const nBLACK10 = '#1F2025'
export const nBLACK30 = '#28292E'
export const nBLACK40 = '#050608'

export const nGRAY10 = '#6B6E79'
export const nGRAY20 = '#858995'
export const nGRAY25 = '#54575e'
export const nGRAY30 = '#2f3137'
export const nGRAY35 = '#52555D'
export const nGRAY40 = '#35363B'


export const tGRAY20 = '#85899599'
export const tGRAY30 = '#545458A6'
export const tGRAY40 = '#45464C'
export const GRAY40 = '#cfcfd0'
export const GRAY45 = '#c9c9cb'
export const GRAY50 = '#aaaaaa'
export const GRAY60 = '#52545D'
export const t40GRAY60 = '#52545D66'
export const GRAY70 = '#383B40'
export const GRAY90 = '#27292d'
export const GRAY100 = '#1f2023'
export const GRAY110 = '#151619'

export const GRAY_ONBOARD = '#8791a5'

export const BLUE = '#50b5e8'
export const BLUE2 = '#0A84FF'

export const PURPLE = '#9A1F99'

export const nBLUE = '#8D88A8'
export const nPURPLE = '#C47AF5'
export const nPURPLE10 = '#873FBB'
export const nPURPLE20 = '#873FBB'
export const nPURPLE40 = '#C882F8'

export const nORANGE10 = '#FF6705'
export const nORANGE20 = '#FF7905'
export const nORANGE40 = '#FFB700'

export const YELLOW2 = '#EBC789'
export const nYELLOW = '#F5D193'

export const nGRAYBACKGROUND2 = '#292A30'
export const nGRAYBACKGROUND3 = '#383940'
export const nGRAYBACKGROUND4 = '#989898'

export const gradient = {
  COLORS: [BLUE, PURPLE],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}

export const orangeGradient = {
  COLORS: [nORANGE10, nORANGE40],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}