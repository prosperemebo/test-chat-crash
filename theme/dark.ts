import {
  WHITE,
  BLACK,
  nGRAY10,
  GRAY_ONBOARD,
  nPURPLE,
  nBLUE,
  GRAY40,
  nGRAY20,
  RED1,
  tGRAY20,
  RED,
  nBLACK,
  nORANGE10,
  nORANGE40,
  nPURPLE10,
  GRAY70,
  tGRAY30,
  BLUE2,
  BLACK2,
  nGRAY25,
  nBLACK30,
  GRAY50,
  nBLACK10,
  t10WHITE,
  nGRAYBACKGROUND2,
  GRAY100,
  GRAY60,
  nGRAY40,
  nGRAY30,
  YELLOW2,
  nYELLOW,
  nGRAYBACKGROUND3,
  tGRAY40,
  nGRAY35,
  nBLACK40,
  t40GRAY60,
  nPURPLE20,
  nPURPLE40,
} from './color'

export const dark = {
  tint: nPURPLE,
  primaryTextColor: WHITE,
  primaryBg: nBLACK10,
  headerBg: GRAY100,
  headerBorderColor: GRAY60,
  secondaryTextColor: GRAY60,
  secondaryBgLight: nBLACK30,
  secondaryBg: nBLACK,
  serviceBg: GRAY100,
  inActiveBorderColor2: t10WHITE,
  backgroundColor2: nGRAYBACKGROUND2,
  backgroundColor3: nGRAYBACKGROUND3,
  backgroundColor4: nBLACK10,

  auth: {
    socialIconColor: WHITE,
    socialLoginButton: BLACK,
    dividerColor: nGRAY10,
    noticeText: WHITE,
    forgotPasswordLabel: WHITE,
  },

  onboarding: {
    headerTextColor: WHITE,
    headerStepsTextColor: GRAY_ONBOARD,
    progressBarColor: nPURPLE,
    progressBarBackgroundColor: nBLUE,
    selectedItemTextColor: WHITE,
    disabledItemTextColor: GRAY40,
    gameSectionOverlayColor: nPURPLE,
    gameEmptySectionColor: BLACK,
    gameEmptyIconColor: BLACK2,
  },

  common: {
    labelColor: WHITE,
    infoTextColor: nGRAY20,
    inputIconColor: nGRAY20,
    inputPlaceholderColor: nGRAY20,
    serviceTextColor: WHITE,
    inputTextColor: WHITE,
    inputTextError: RED1,
    inputBorderColor: tGRAY20,
    inputErrorBorderColor: RED,
    inputBackgroundColor: nBLACK,
    toastBackground: nBLACK,
    toastBorderColor: nPURPLE,
    toastTextColor: WHITE,
    buttonGradientOrangeStart: nORANGE10,
    buttonGradientOrangeEnd: nORANGE40,
    buttonGradientPurpleStart: nPURPLE40,
    buttonGradientPurpleEnd: nPURPLE20,
    buttonGradientGrayStart: nGRAY10,
    buttonGradientGrayEnd: nGRAY20,
    buttonDartBackgroundColor: BLACK,
    backButtonColor: nGRAY20,
    switchTextColor: WHITE,
    checkboxIconColor: nPURPLE,
    gradientButtonColorStart: nPURPLE10,
    gradientButtonColorEnd: nPURPLE,
    borderColor: GRAY70,
    genderPickerBorderColor: nGRAY25,
    genderPickerActiveTextColor: WHITE,
    genderPickerTextColor: GRAY50,
    checkboxTextColor: WHITE,
    iconButtonColor: WHITE,
    backgroundColor: nBLACK10,
  },

  settings: {
    socialIconBackgroundColor: nGRAY40,
    premiumFeatureTextColor: nPURPLE,
    buttonIconColor: nGRAY20,
    serviceItemBgColor: nGRAY30,
  },

  postOnboarding: {
    backgroundColor: nBLACK10,
    inputIconColor: nGRAY20,
    inputIconErrorColor: RED1,
    disclosureTextColor: nGRAY20,
    loadingIndicatorColor: BLACK,
  },

  profile: {
    backgroundColor: nBLACK10,
    improveBackgroundColor: YELLOW2,
    improveBannerBackgroundColor: nYELLOW,
    improveBannerTitleColor: nBLACK30,
    coverPictureCardRemoveButtonBg: tGRAY30,
    profilePicturesCountColor: tGRAY40,
  },

  chats: {
    iconColor: nGRAY20,
    inputColor: nBLACK40,
    placeholderColor: nGRAY35,
    appbarBorderColor: t40GRAY60,
    backgroundColor: nBLACK10,
  },

  nav: {
    bottomTabLabelBackgroundColor: nBLACK,
    bottomTabBackgroundColor: nBLACK,
  },

  other: {
    updateApp: {
      textBorderColor: tGRAY30,
      buttonTextColor: BLUE2,
      textColor: WHITE,
    },
  },
}
