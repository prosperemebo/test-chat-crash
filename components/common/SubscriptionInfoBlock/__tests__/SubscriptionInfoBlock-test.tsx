import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { describe, expect, test } from '@jest/globals'
import SubscriptionInfoBlock from '../'
import { dark } from '@/theme'

describe('<SubscriptionInfoBlock />', () => {

  test('SubscriptionInfoBlock renders correctly', () => {
    const tree = render(
      <ThemeProvider theme={dark}>
        <SubscriptionInfoBlock from={'settings'} hasMembership={true}/>
        <SubscriptionInfoBlock from={'settings'} hasMembership={false}/>
      </ThemeProvider>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})