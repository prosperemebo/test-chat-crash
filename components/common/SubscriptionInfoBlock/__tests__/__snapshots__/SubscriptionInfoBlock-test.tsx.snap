// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubscriptionInfoBlock /> SubscriptionInfoBlock renders correctly 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "marginHorizontal": 15,
        "opacity": 1,
      }
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      colors={
        [
          "#873FBB",
          "#C47AF5",
        ]
      }
      end={
        {
          "x": 1,
          "y": 0,
        }
      }
      start={
        {
          "x": 0,
          "y": 0,
        }
      }
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
          "overflow": "hidden",
          "width": 350,
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "Poppins-Bold",
              "fontSize": 18,
              "lineHeight": 27,
              "textAlign": "center",
            }
          }
        >
          MOCKED_STRING:subscription:manageSubscriptionsText
        </Text>
      </View>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "marginHorizontal": 15,
        "opacity": 1,
      }
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      colors={
        [
          "#873FBB",
          "#C47AF5",
        ]
      }
      end={
        {
          "x": 1,
          "y": 0,
        }
      }
      start={
        {
          "x": 0,
          "y": 0,
        }
      }
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
          "overflow": "hidden",
          "width": 350,
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "center",
              "width": 250,
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontFamily": "Poppins-Bold",
                "fontSize": 18,
                "lineHeight": 27,
                "textAlign": "center",
              }
            }
          >
            MOCKED_STRING:subscription:unlockAllBenefitsHeader
          </Text>
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontFamily": "Poppins-Light",
                "fontSize": 12,
                "letterSpacing": 0,
                "lineHeight": 17,
                "marginLeft": 4,
                "opacity": 0.8,
                "textAlign": "center",
              }
            }
          >
            MOCKED_STRING:subscription:unlockAllBenefitsDescription
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#28292E",
                "borderBottomLeftRadius": 50,
                "borderBottomRightRadius": 50,
                "borderColor": "#28292E",
                "borderStyle": "solid",
                "borderTopLeftRadius": 50,
                "borderTopRightRadius": 50,
                "borderWidth": 1,
                "gap": 1,
                "height": 40,
                "marginBottom": 10,
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 10,
                "opacity": 1,
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 10,
                "width": 130,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Poppins-Bold",
                  "fontSize": 12,
                  "lineHeight": 18,
                  "textAlign": "center",
                }
              }
            >
              MOCKED_STRING:from $4.99/mo
            </Text>
          </View>
        </View>
      </View>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>,
]
`;
